src/LiveComponent/src/Util/LiveControllerAttributesCreator.php

```diff
@@ -57,7 +57,11 @@ public function attributesForRendering(MountedComponent $mounted, ComponentMetad
         $attributesCollection = $this->attributeHelper->create();
         $attributesCollection->setLiveController($mounted->getName());
 
-        $url = $this->urlGenerator->generate($metadata->get('route'), ['_live_component' => $mounted->getName()], $metadata->get('url_reference_type'));
+        $url = $this->urlGenerator->generate(
+            $metadata->get('route') ?? 'ux_live_component',
+            ['_live_component' => $mounted->getName()],
+            $metadata->get('url_reference_type') ?? UrlGeneratorInterface::ABSOLUTE_PATH
```

Comment:
Why this change?

Reply:
I'm not sure I understand the question (since I explained why in the PR description).

If we manually declare the service in the YAML file, we have to provide this additional information.

These default internal values ​​are precisely designed to prevent this.